### ISSUE TITLE
ProductProjectionPagedQueryResponse is Generated with a long Offset while the query only accepts a int32 offset.

### ISSUE THREAD

**Issue by jwdb** [commercetools/commercetools-dotnet-core-sdk-v2#269]
**Describe the bug**
When requesting the Product projection in a paginated manner, the returned offset is a long (int64) while the request can only be performed with an int (int32).

**To Reproduce**
Reuse the resulting Offset from the response to determine the new offset to query.

**Expected behavior**
I expect the resulting offset to be in the same type as is needed to query.

**Screenshots/Code snippet**

```csharp
int currentOffset = 0;
IProductProjectionPagedQueryResponse? products = await projectApiRoot
    .ProductProjections()
    .Get()
    .WithOffset(currentOffset)
    .WithLimit(100)
    .ExecuteAsync();

while (products.Total < currentOffset)
{
    products = await projectApiRoot
        .ProductProjections()
        .Get()
        .WithOffset(currentOffset)
        .WithLimit(100)
        .ExecuteAsync();

    currentOffset = products.Offset + 100; // this line throws error CS0266: Cannot implicitly convert type 'long' to 'int'. An explicit conversion exists (are you missing a cast?)
}
```

**Stack information (please complete the following information):**
 - dotnet: 8
 - SDK: 11.2.0

**Additional context**
The documentation also states the Offset should both be an `int`: https://docs.commercetools.com/api/projects/productProjections#ctp:api:type:ProductProjectionPagedQueryResponse

**Comment by jenschude** [comment]
That's a bummer.

Looking into it, there is no simple solution. We will take a closer look to find a way to best deal with this in a backwards compatible way.

**Comment by jenschude** [comment]
Released 11.4.0 which now accepts long values and should be backwards compatible as int values are also valid.
